src/components/internals/localized-input-toggle/localized-input-toggle.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import FlatButton from '@commercetools-uikit/flat-button';
+import { WorldIcon } from '@commercetools-uikit/icons';
+import messages from '../messages/localized-input/messages';
+
+const LocalizedInputToggle = (props) => {
+  const intl = useIntl();
+
+  const labelMessage = props.isOpen ? props.hideMessage : props.showMessage;
+  const label =
+    typeof labelMessage === 'string'
+      ? labelMessage
+      : intl.formatMessage(labelMessage, {
+          remainingLanguages: props.remainingLocalizations,
+        });
+
+  return (
+    <FlatButton
+      icon={props.icon}
+      label={label}
+      onClick={props.onClick}
+      isDisabled={props.isDisabled}
+    />
+  );
+};
+
+const intlMessageShape = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  description: PropTypes.string,
+  defaultMessage: PropTypes.string.isRequired,
+});
+
+LocalizedInputToggle.displayName = 'LocalizedInputToggle';
+LocalizedInputToggle.propTypes = {
+  icon: PropTypes.node.isRequired,
+  isOpen: PropTypes.bool,
+  onClick: PropTypes.func.isRequired,
+  isDisabled: PropTypes.bool,
+  showMessage: PropTypes.oneOfType([PropTypes.string, intlMessageShape])
+    .isRequired,
+  hideMessage: PropTypes.oneOfType([PropTypes.string, intlMessageShape])
+    .isRequired,
+  remainingLocalizations: PropTypes.number,
+};
+LocalizedInputToggle.defaultProps = {
+  icon: <WorldIcon />,
```

Comment:
I don't think we can do this here. Just do the fall back inline where you need to pass the prop.

Reply:
Ok, is this what you meant https://github.com/commercetools/ui-kit/pull/1590/commits/79a01bba21942801ab2aa1644195f40cebe8379d?

Reply:
Hm. why can't we do it? It's actually an interesting question :)